features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx

```diff
@@ -0,0 +1,471 @@
+import MUIDataTable from 'mui-datatables';
+import ContentBase from 'AppComponents/AdminPages/Addons/ContentBase';
+import InlineProgress from 'AppComponents/AdminPages/Addons/InlineProgress';
+import Alert from 'AppComponents/Shared/Alert';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import HelpBase from 'AppComponents/AdminPages/Addons/HelpBase';
+import DescriptionIcon from '@material-ui/icons/Description';
+import Link from '@material-ui/core/Link';
+import Configurations from 'Config';
+import API from 'AppData/api';
+import Button from '@material-ui/core/Button';
+import * as dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import localizedFormat from 'dayjs/plugin/localizedFormat';
+import CheckIcon from '@material-ui/icons/Check';
+import ClearIcon from '@material-ui/icons/Clear';
+import Box from '@material-ui/core/Box';
+
+const useStyles = makeStyles((theme) => ({
+    searchBar: {
+        borderBottom: '1px solid rgba(0, 0, 0, 0.12)',
+    },
+    searchInput: {
+        fontSize: theme.typography.fontSize,
+    },
+    block: {
+        display: 'block',
+    },
+    contentWrapper: {
+        margin: '40px 16px',
+    },
+    button: {
+        borderColor: 'rgba(255, 255, 255, 0.7)',
+    },
+    approveButton: {
+        textDecoration: 'none',
+        backgroundColor: theme.palette.success.light,
+    },
+    rejectButton: {
+        textDecoration: 'none',
+        backgroundColor: theme.palette.error.light,
+    },
+}));
+
+/**
+ * Render a list
+ * @param {JSON} props props passed from parent
+ * @returns {JSX} Header AppBar components.
+ */
+function ListLabels() {
+    const intl = useIntl();
+    const [data, setData] = useState(null);
+    const restApi = new API();
+
+    /**
+     * API call to get Detected Data
+     * @returns {Promise}.
+     */
+    function apiCall() {
+        return restApi
+            .workflowsGet('AM_API_STATE')
+            .then((result) => {
+                const workflowlist = result.body.list;
+                return (workflowlist);
+            })
+            .catch((error) => {
+                throw (error);
+            });
+    }
+
+    const fetchData = () => {
+    // Fetch data from backend
+        setData(null);
+        const promiseAPICall = apiCall();
+        promiseAPICall.then((LocalData) => {
+            setData(LocalData);
+        })
+            .catch((e) => {
+                Alert.error(e);
+            });
+    };
+
+    useEffect(() => {
+        fetchData();
+    }, []);
+
+
+    const updateStatus = (referenceId, value) => {
+        const body = {};
+        body.status = value;
+        body.attributes = {};
+        body.description = 'Approve workflow request.';
+
+        let promisedupdateWorkflow = '';
```

Comment:
we can make promisedupdateWorkflow a const right?

const promisedupdateWorkflow = restApi.updateWorkflow(referenceId, body);

Reply:
fixed with 54dd617